.github/workflows/windows-msvc.yml

```diff
@@ -0,0 +1,125 @@
+name: Windows MSVC Workflow
+
+on:
+ pull_request:
+   branches: [ gcos4gnucobol-3.x ]
+ push:
+   branches: [ gcos4gnucobol-3.x ]
+
+env:
+  FLEXBISON: https://github.com/lexxmark/winflexbison/releases/download/v2.5.25/win_flex_bison-2.5.25.zip
+  MSBUILD: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\MSBuild.exe
+  VCPKG_ROOT: C:\vcpkg
+  VCPKGS: mpir:x64-windows pdcurses:x64-windows berkeleydb:x64-windows libxml2:x64-windows cjson:x64-windows
+  MSYS2_ROOT: C:\msys64
+  MSYSPKGS: autoconf
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - windows-latest
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+
+      - name: Set git user
+        run: |
+          git config --global user.name github-actions
+          git config --global user.email github-actions-bot@users.noreply.github.com
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Restore VCPKG package cache
+        uses: actions/cache/restore@v3
+        id: restore-vcpkg-cache
+        with:
+          key: cache-vcpkg
+    
+      - name: Install VCPKG packages
+        if: steps.restore-vcpkg-cache.outputs.cache-hit != 'true'
```

Comment:
You likely either want to run that in any case (should be quite fast after getting the cache) or want to do that once a day/week, so you do get updates also without manually dropping the cache.